engine/apps/webhooks/utils.py

```diff
@@ -124,6 +125,18 @@ def _serialize_event_user(user):
     }
 
 
+def _extract_users_from_escalation_snapshot(escalation_snapshot):
+    users = []
+    if escalation_snapshot:
+        for policy_snapshot in escalation_snapshot.escalation_policies_snapshots:
+            for user in policy_snapshot.notify_to_users_queue:
+                users.append(_serialize_event_user(user))
+            if policy_snapshot.notify_schedule:
+                for user in list_users_to_notify_from_ical(policy_snapshot.notify_schedule):
+                    users.append(_serialize_event_user(user))
+    return list({u["id"]: u for u in users}.values())
```

Comment:
Maybe build a set from the beginning, adding the serialized user to it, and then return the list of the set?

Reply:
Not sure if it is correct but I was trying to maintain insertion order as well as eliminating duplicates, I don't think I can do that with a set?  Reason behind it was if the escalation snapshot has multiple steps that it can get users from we would have a more accurate order.  My assumption is generally it will be more useful to get the first element for our best guess at "next" rather than searching through the array.

Reply:
Oh, I see. In that case, are we interested in knowing if a person will be notified multiple times? (and then just having a list where we append users?)

Reply:
Good catch, I'll fix it in the current webhooks PR, something like
`list({u["foo"]: u for u in users if u}.values()' shdould cover it